Change `EventDispatcher.shutdownWithProtocolError` to throw, not return

gavin-norman-sociomantic opened the pull request:
The returned exception is intended to always be thrown, and it would be a
bug, really, if the caller did not do so. Therefore, making the method
throw directly simplifies usage and removes one vector for bugs.

nemanja-boric-sociomantic commented on src/swarm/neo/connection/RequestOnConnBase.d:
```diff
@@ -528,7 +527,7 @@ abstract class RequestOnConnBase
             auto e = this.outer.connection.protocol_error.set(msg, file, line);
             this.shutdownConnection(e);
-            return e;
+            throw e;
```
`void shutdownWithProtocolError`

gavin-norman-sociomantic replied:
Haha yeah. Updated.

Commit:
Change `EventDispatcher.shutdownWithProtocolError` to throw, not return

The returned exception is intended to always be thrown, and it would be a
bug, really, if the caller did not do so. Therefore, making the method
throw directly simplifies usage and removes one vector for bugs.
